commit 0c86abf954ca4a5f00c26f6600acac93f9fc3538
Author: Robin Watts <robin.watts@artifex.com>
Date:   Tue Feb 28 17:15:40 2017 +0000

    Bug 697395: Fix underflow in special case scaler.
    
    When scaling a single row pixmap with a flip, I was getting
    the offset to the far end of the line wrong due to forgetting
    to allow for the alpha plane.
    
    Fixed here.

diff --git a/source/fitz/draw-scale-simple.c b/source/fitz/draw-scale-simple.c
index 1525c7b..1d59092 100644
--- a/source/fitz/draw-scale-simple.c
+++ b/source/fitz/draw-scale-simple.c
@@ -1294,7 +1294,7 @@ scale_single_row(unsigned char * restrict dst, int dstride, const unsigned char
 		tmp[j] = 128;
 	if (weights->flip)
 	{
-		dst += (weights->count-1)*n;
+		dst += (weights->count-1)*nf;
 		for (i=weights->count; i > 0; i--)
 		{
 			min = *contrib++;
